packages/shared/args.js

```diff
@@ -0,0 +1,30 @@
+function args(args) {
+    const parsedArgs = {
+        named: { },
+        positional: []
+    };
+    const [...rest] = args;
+
+    rest.forEach((arg, index) => {
+        if (arg.indexOf("-") != -1) {
+            const name = getArgName(arg)
+            parsedArgs.named[name] = rest[index + 1];
+            rest.shift();
```

Comment:
In general, mutating a the object you are iterating over is considered dangerous. And I guess you did a shallow copy of args in order to not mutate the input parameter. So i might think that a standard for loop over args would be better.